generator/tree/builder.py

```diff
@@ -113,6 +117,33 @@ def _append_constant_references(root):
         for c in constants:
             a.insert(ConstantReference(c.path))
 
+def _update_field_type_references(root):
+    fields = [f for f in root.iterate(nodes.Field)]
```

Comment:
No need to create a list, a generator expression should be enough:
```
(f for f in root.iterate(nodes.Field))
```

Actually, why not just to write `f in root.iterate(nodes.Field)`?